src/Authentication/Actions/Email2FA.php

```diff
@@ -23,17 +23,17 @@ public function show()
     {
         $user = auth()->user();
 
+        /** @var UserIdentityModel $identityModel */
+        $identityModel = model(UserIdentityModel::class);
+
         // Delete any previous activation identities
-        $identities = new UserIdentityModel();
-        $identities->where('user_id', $user->getAuthId())
-            ->where('type', 'email_2fa')
-            ->delete();
+        $identityModel->deleteIdentitiesByType($user->getAuthId(), 'email_2fa');
```

Comment:
In general I like the plural noun for the model variable, like `$identities` - it makes the code read smoothly. Not opposed to the more utilitarian version including "Model", but a style consideration we should probably be consistent with.

Reply:
`$identities` implies multiple identities, and there was the code to get multiple identities from the Model.
So I used `$identityModel` in the all places.

```php
        $identities = $identityModel->getIdentitiesByTypes(
            auth('session')->id(),
            ['email_2fa', 'email_activate']
        );
```

Reply:
Acknowledged.